src/main/java/com/crud/tasks/service/DbService.java

```diff
@@ -0,0 +1,28 @@
+package com.crud.tasks.service;
+
+import com.crud.tasks.domain.Task;
+import com.crud.tasks.repository.TaskRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DbService {
+
+    @Autowired
+    private TaskRepository repository;
+
+    public List<Task> getAllTasks() {
+        return repository.findAll();
+    }
+
+    public Task getTaskById(Long id) {
+        Optional<Task> matchingTask = repository.findAll().stream()
+                .filter(task -> task.getId()==id)
+                .findFirst();
```

Comment:
to rozwiazanie jest poprawne. Jesli w bazie badą miliony taskow, aplikacji skonczy sie pamiec. Musisz napisac nowa metode w `TaskRepository` ktora wyciaga tylko po id

Reply:
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/

Reply:
czy poniższe będzie w takim razie ok?

public Task getTaskById(Long id) {
        Optional<Task> matchingTask = repository.findById(id).stream().findAny();
        return matchingTask.orElse(null);
    }

Reply:
to jest nieco bez sensu, poniewaz findById zwraca i tak optionala

Reply:
mozesz napisac albo
return repository.findById(id).orElse(null); //gorsze rozwiazanie
return repository.findById(id).orElseThrow(()->TaskNotFoundException());//gdzie TaskNotFoundEx.. to twoj customowy exception rozszezajacy runtimeexception